### PR TITLE
ENP-292 Upgrade microsoft coding test

### PR DESCRIPTION
**Pull request by pradhanangan**
**JIRA:** https://clearpoint.atlassian.net/browse/ENP-292

**Changes:**
Frontend:
Upgraded .NET target framework to 7.0
Nuget packages upgraded to latest stable version

Backend:
Upgraded to react 18 and dependencies to latest

### CODE REVIEW

**Review by michaelm-clearpoint — DISMISSED** [review]

File: Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj [hunk 1]
```diff
@@ -1,19 +1,19 @@
-<Project Sdk="Microsoft.NET.Sdk">
+﻿<Project Sdk="Microsoft.NET.Sdk">
 
   <PropertyGroup>
-    <TargetFramework>net5.0</TargetFramework>
+    <TargetFramework>net7.0</TargetFramework>
```
Comment (michaelm-clearpoint) [inline Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj:4]: maybe it's good to leave the version to an outdated one that is still supported by Microsoft (.net 6?), so candidates can showcase extra skills if they update the project to latest version?

Comment (pradhanangan) [reply Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj:4]: @michaelm-clearpoint shall I change target framework to .NET 6.

Comment (michaelm-clearpoint) [reply Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj:4]: yes for me, what you think @mhurbuns @mniem @MikeNguyenCP @gowri-cp ?

Comment (gowri-cp) [reply Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj:4]: yes,  its a good way to check if the candidate will upgrade or not

Comment (pradhanangan) [reply Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj]: I have changed to .NET 6.0

**Review by michaelm-clearpoint — APPROVED** [review]

File: Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj (hunk 1 above)
Comment (morne-niemand) [reply Backend/TodoList.Api/TodoList.Api.UnitTests/TodoList.Api.UnitTests.csproj]: LGTM

**Review by MikeNguyenCP — APPROVED** [review]
LGTM

**Pull request merged**